src/EnergyPlus/AirLoopHVACDOAS.cc

```diff
@@ -1018,28 +1018,28 @@ namespace AirLoopHVACDOAS {
 
     void AirLoopDOAS::GetDesignDayConditions(EnergyPlusData &state)
     {
-
-        // int constexpr summerDesignDayTypeIndex(9);
-        // int constexpr winterDesignDayTypeIndex(10);
-        for (size_t env = 1; env <= state.dataWeatherManager->Environment.size(); ++env) {
-            switch (state.dataWeatherManager->Environment(env).KindOfEnvrn) {
-            case DataGlobalConstants::KindOfSim::DesignDay:
-            case DataGlobalConstants::KindOfSim::RunPeriodDesign: {
-                if (state.dataWeatherManager->Environment(env).maxCoolingOATSizing > this->SizingCoolOATemp) {
-                    this->SizingCoolOATemp = state.dataWeatherManager->Environment(env).maxCoolingOATSizing;
+        for (auto &env : state.dataWeatherManager->Environment) {
+            if (env.KindOfEnvrn != DataGlobalConstants::KindOfSim::DesignDay && env.KindOfEnvrn != DataGlobalConstants::KindOfSim::RunPeriodDesign)
+                continue;
+            if (env.maxCoolingOATSizing > this->SizingCoolOATemp) {
+                this->SizingCoolOATemp = env.maxCoolingOATSizing;
+                if (env.KindOfEnvrn == DataGlobalConstants::KindOfSim::DesignDay &&
+                    state.dataWeatherManager->DesDayInput(env.DesignDayNum).PressureEntered) {
```

Comment:
I verified with a modified example file, SmallOffice_CentralDOAS_wWeatherDesignDay (attached to last PR comment), that (env.DesignDayNum) does = 0 for SizingPeriod:WeatherFileDays (and is non-0 for design days) and that this line does not crash due to a 0 index in DesDayInput(0).PressureEntered.